### PR TITLE
[Enhancement] Use log4j to control hudi logs

### PR DESCRIPTION
**Pull request by miomiocat**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by miomiocat** [comment]
run starrocks_admit_test

File: java-extensions/hudi-reader/src/main/java/com/starrocks/hudi/reader/HudiSliceScanner.java [hunk 1]
```diff
@@ -201,7 +205,7 @@ public void open() throws IOException {
             initReader(jobConf, properties);
         } catch (Exception e) {
             close();
-            e.printStackTrace();
+            LOG.error("Failed to open the hudi MOR slice reader.", e);
```
Comment (stephen-shelby) [inline java-extensions/hudi-reader/src/main/java/com/starrocks/hudi/reader/HudiSliceScanner.java:208]: make it to a constant variable?

**Review by stephen-shelby — DISMISSED** [review]

**Review by Youngwb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
@Mergify rebase

**Comment by wanpengfei-git** [comment]
@mergify rebase

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**